frontend/src/actionCreators/partiesActionCreator.js

```diff
@@ -203,12 +175,16 @@ export const updatePartyRelationship = (payload) => (dispatch) => {
     });
 };
 
-export const fetchPartyRelationships = (mineId) => (dispatch) => {
+export const fetchPartyRelationships = (mineId, partyId, typeCode) => (dispatch) => {
   dispatch(request(reducerTypes.FETCH_PARTY_RELATIONSHIPS));
-  dispatch(showLoading("modal"));
+  dispatch(showLoading());
   return axios
     .get(
-      `${ENVIRONMENT.apiUrl + API.PARTY_RELATIONSHIP}?mine_guid=${mineId}`,
+      `${ENVIRONMENT.apiUrl + API.PARTY_RELATIONSHIP}?${
+        mineId ? `mine_guid=${mineId}${partyId || typeCode ? "&" : ""}` : ""
+      }${partyId ? `party_guid=${partyId}${typeCode ? "&" : ""}` : ""}${
+        typeCode ? `types=${typeCode}` : ""
```

Comment:
The `query-string` library takes care of all this logic, and we're already using it. You can see an example here:
https://github.com/bcgov/mds/blob/develop/frontend/src/constants/routes.js#L15

It accepts an option of the query params, where the key is the param name and the value is the param value.

Reply:
Nice, I wasn't aware of this.